metricbeat/module/jolokia/jmx/jmx.go

```diff
@@ -0,0 +1,98 @@
+package jmx
+
+import (
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/metricbeat/helper"
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/metricbeat/mb/parse"
+)
+
+var (
+	debugf = logp.MakeDebug("jolokia-jmx")
+)
+
+// init registers the MetricSet with the central registry.
+func init() {
+	if err := mb.Registry.AddMetricSet("jolokia", "jmx", New, hostParser); err != nil {
+		panic(err)
+	}
+}
+
+const (
+	// defaultScheme is the default scheme to use when it is not specified in
+	// the host config.
+	defaultScheme = "http"
+
+	// defaultPath is the default path to the ngx_http_stub_status_module endpoint on Nginx.
+	defaultPath = "/jolokia/?ignoreErrors=true&canonicalNaming=false"
+)
+
+var (
+	hostParser = parse.URLHostParserBuilder{
+		DefaultScheme: defaultScheme,
+		PathConfigKey: "path",
+		DefaultPath:   defaultPath,
+	}.Build()
+)
+
+// MetricSet type defines all fields of the MetricSet
+type MetricSet struct {
+	mb.BaseMetricSet
+	metricSetConfig MetricSetConfig // array containing urls, bodies and mappings
+	namespace       string
+	http            *helper.HTTP
+}
+
+// New create a new instance of the MetricSet
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	logp.Warn("BETA: The jolokia jmx metricset is beta")
+
+	// Additional configuration options
+	config := struct {
+		Namespace string        `config:"namespace" validate:"required"`
+		Mapping   []MetricSetup `config:"jmx.mappings" validate:"required"`
+		// TODO: What are these settings for exactly?
```

Comment:
I'm still in progress of figuring this one out: https://github.com/elastic/beats/pull/3051#issuecomment-279323726 I added the `in progress` label again to the PR to make sure it does not get merged by accident.